src/org/omegat/core/dictionaries/StarDict.java

```diff
@@ -247,16 +250,17 @@ private synchronized String readArticle(int start, int len) {
          */
         private String readDictArticleText(int start, int len) {
             String result = null;
-            try (FileInputStream in = new FileInputStream(dataFile)) {
-                byte[] data = new byte[len];
-                long moved = in.skip(start);
-                if (moved < start) {
-                    long moved2 = in.skip(start - moved);
-                    if (moved2 < start - moved) {
-                        throw new IOException("Cannot seek dictionary.");
-                    }
+            if(dictFile == null){
+                try {
+                    dictFile = new RandomAccessFile(new File(dataFile), "r");
+                } catch (FileNotFoundException e){
```

Comment:
Please follow project whitespace conventions.

```suggestion
            if (dictFile == null) {
                try {
                    dictFile = new RandomAccessFile(new File(dataFile), "r");
                } catch (FileNotFoundException e) {
```

Reply:
However I am not sure about instantiating the file here. If a user has loaded a dictionary, I think it's OK to assume they want to use it. So I would unconditionally instantiate the `RandomAccessFile` elsewhere, like in the constructor.

Reply:
I think you're right.
I will move instantiate to constructor.
Thank you.

Reply:
Thanks